src/hotspot/share/gc/shared/gcConfig.cpp

```diff
@@ -111,6 +111,38 @@ void GCConfig::select_gc_ergonomically() {
   }
 }
 
+void select_gc_ergonomically_dedicated() {
```

Comment:
Are we missing Shenandoah?

Reply:
I've been getting some response on using ZGC only for 32+ GB.

I think as we go for ergonomics defaults, it's either ZGC or Shenandoah for large heaps. Either one should be fine as a default large heap GC, in my opinion.

Reply:
I am curious how folks are thinking about handling larger memory and heaps and if there is a future follow up to set an `intent` for a given JVM, for example `throughput` versus `low-pause` In addition to `dedicated` versus `shared`. This would add more configuration complexity (something this JEP & PR intends to reduce), but could allow for better ergonomic heuristics targeting a given environment's resources.

The recent removal of ZGC from the initial ergonomics profile simplifies the heap memory math as ZGC has an additional fixed 3% non-heap native memory overhead, and is one place we have had to adjust compared to G1 and Shenandoah GC, especially in shared containerized environments with significant off-heap memory usage (direct ByteBuffers, native compression libraries, some data processing libraries like parquet, arrow, SQLite, etc.).

https://netflixtechblog.com/bending-pause-times-to-your-will-with-generational-zgc-256629c9386b

> ZGC has a fixed overhead 3% of the heap size, requiring more native memory than G1. Except in a couple of cases, there’s been no need to lower the maximum heap size to allow for more headroom, and those were services with greater than average native memory needs.